### PR TITLE
WIP: Wrap gc suballocator size check in CompressedPointers scope

### PR DESCRIPTION
**Pull request by fengxue-IS**
Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

File: runtime/gc_modron_startup/mminit.cpp [hunk 1]
```diff
@@ -1558,7 +1558,9 @@ combinationMemoryParameterVerification(J9JavaVM *javaVM, IDATA* memoryParameters
 	bool opt_XmosSet = (-1 != memoryParameters[opt_Xmos]);
 	bool opt_XmoxSet = (-1 != memoryParameters[opt_Xmox]);
 	bool opt_XsoftmxSet = (-1 != memoryParameters[opt_Xsoftmx]);
+#if defined(J9VM_GC_COMPRESSED_POINTERS)
```
Comment (charliegracie) [inline runtime/gc_modron_startup/mminit.cpp]: What is the documentation for this option?

**Comment by dmitripivkine** [comment]
What the reason for this change? Suballocator as feature is not limited (but happen to be used only) to Compressed Refs spec currently.

File: runtime/gc_modron_startup/mminit.cpp (hunk 1 above)
Comment (dmitripivkine) [reply runtime/gc_modron_startup/mminit.cpp]:
```
Sets an initial size for an area in memory that is reserved for any native classes, monitors, and threads that are used by compressed references within the lowest 4 GB memory area.
```
https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/openj9/xmcrs/index.html

**Comment by fengxue-IS** [comment]
@dmitripivkine The suballocator values are currently only initialized for compressed refs specs in OMR GC library, this check however runs unconditionally which fails when running with -Xcheck:memory.

I proposed this change in openj9 since OMR only used the suballocator values within compressed refs scope. (therefore correct by it self)

**Comment by dmitripivkine** [comment]
>The suballocator values are currently only initialized for compressed refs specs in OMR GC library, this check however runs unconditionally which fails when running with -Xcheck:memory.
> 
Please provide more details how it relates to failure when running with -Xcheck:memory

**Comment by dmitripivkine** [comment]
ok, I see where the problem is:
in OMR in GCExtensionsBase.hpp
```
	uintptr_t suballocatorInitialSize; /**< the initial chunk size in bytes for the J9Heap suballocator (enabled with the -Xgc:suballocatorInitialSize option) */
	uintptr_t suballocatorCommitSize; /**< the commit size in bytes for the J9Heap suballocator (enabled with the -Xgc:suballocatorCommitSize option) */
```
are defined unconditionally however an initialization of these variables in Constructor done for Compressed Refs only:
```
#if defined(OMR_GC_COMPRESSED_POINTERS)
		, suballocatorInitialSize(SUBALLOCATOR_INITIAL_SIZE) /* default for J9Heap suballocator initial size is 200 MB */
		, suballocatorCommitSize(SUBALLOCATOR_COMMIT_SIZE) /* default for J9Heap suballocator commit size is 50 MB */
```
So in my opinion proposed fix is a hack and it is not good enough. 
We should fix root of the problem and make code consistent. In order to do this we should make initialization of these variables in Constructor unconditional for all platforms () which require definitions for SUBALLOCATOR_INITIAL_SIZE and SUBALLOCATOR_COMMIT_SIZE be for all platforms too (or alternatively set both to 0). Another possibility is opposite: make variables be defined for CR only (and as result add missed define checks everywhere missed). 
I am voting for first solution because it makes code more general.
@charliegracie What is your opinion?

**Comment by charliegracie** [comment]
-Xcheck:memory fails because those two values are not initialized?  Can you provide more details here?

**Comment by dmitripivkine** [comment]
Fixed at OMR side  eclipse/omr#3250

**Pull request closed without merging**